### PR TITLE
Prepare 1.0.0-preview4 version of tools

### PR DESCRIPTION
**Pull request by natemcmaster**
cc @pranavkm

### CODE REVIEW

**Comment by natemcmaster** [comment]
:up: :date:

**Comment by natemcmaster** [comment]
We don't want to build aspnet/UserSecrets in the next patch release since Microsoft.Extensions.Configuration.UserSecrets/1.0.1 was moved to aspnet/Configuration, so we'll move dotnet-user-secrets 1.0.0-preview4-final to this repo.

File: src/Microsoft.DotNet.Watcher.Core/project.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "Microsoft.DotNet.ProjectModel": "1.0.0-rc3-003121",
     "Microsoft.Extensions.FileProviders.Abstractions": "1.0.0",
```
Comment (natemcmaster) [inline src/Microsoft.DotNet.Watcher.Core/project.json]: @JunTaoLuo is aspnet/FileSystem getting bumped too?

Comment (JunTaoLuo) [inline src/Microsoft.DotNet.Watcher.Core/project.json]: yes, that repo has been updated in rel/1.0.1

**Pull request merged**